lib/fastly/customer.rb

```diff
@@ -18,9 +29,21 @@ class Customer < Base
     #
     # The id of the user that owns this customer
 
-    # Return a user object representing the owner of this Customer
-    def owner
-      fetcher.get(User, owner_id)
+    ##
+    # :attr: security_contact_id
+    #
+    # The id of the user to be contacted for security issues.
+
+    ##
+    # :attr: technical_contact_id
+    #
+    # The id of the user to be contacted for technical issues.
+
+    [ :billing_contact, :legal_contact, :owner, :security_contact, :technical_contact ].each do |method_name|
+      define_method method_name do
+        id = send("#{method_name}_id")
+        id ? fetcher.get(User, id) : nil
+      end
```

Comment:
Is this sort of meta-programming 🆒 ?

Reply:
It might be nicer (and rdoc would better understand?) if they were separate methods instead of meta-programmed. Could do a private `get_user(id)` or `get_user(attr_name)` function to DRY up the fetcher part at least.

`rake rdoc` should generate them locally if you want to peruse and make sure it looks reasonable for the changes to Customer.

Reply:
🆗